encoding/constructor_temp.go

```diff
@@ -0,0 +1,117 @@
+// Temporary file to hold constructors to ease merge conflicts.
+
+package encoding
+
+import "io"
+
+// STRING
+
+type stringEncoder struct{}
+
+// NewStringEncoder returns a new StringEncoder instance.
+func NewStringEncoder() StringEncoder {
```

Comment:
I think the ctor should take an `io.Writer` no?

Reply:
The `Encode()` call takes to writer. Otherwise we'd need a method to replace the `io.Writer` target.

Reply:
Yeah I like that better, have a both the ctor take an `io.Writer` and a `Reset(io.Writer)` method. Feels a bit strange IMO that the `Encode` call needs to take an `io.Writer`.